### PR TITLE
Fix several issues in geoviewer

### LINKED ISSUE #430: Make `interpolation="None"` the default in `Raster.show()`
By default, `imshow` through `rshow` of Rasterio is interpolating the data, resulting in a LOT of added gaps when there are nodata values.
It's not straightforward to know that you need to pass `interpolation="None"` to remove that effect, it should probably be the default!

### PR DESCRIPTION
**Pull request by adehecq**
This PR solves several inconsistencies/bugs in geoviewer.py:
- solves #412 - only requested bands are now loaded, instead of the selection being made on plotting
- solves #417 - nodata values are taken into account when calculating data percentile
- clarifies the behavior of the tool for multiband raster, as it was unclear and even erroneous (see #412)
- improves tests for single band rasters and add new tests for 3 band rasters
- add a CLI section in the documentation. CLI arguments are automatically generated using "sphinx-argparse" (added as dependency)
- in the same time, I also fixed #430

Left to do:
- add tests to ensure that error is raised for rasters that are not 1 or 3 bands. Added issue #434.

### CODE REVIEW

**Review by rhugonnet — APPROVED** [review]
Looking good, thanks for diving into this!

For documenting: I would add a "command line user guide" like in Rasterio (but in the Documentation section, not as its own header): https://rasterio.readthedocs.io/en/stable/cli.html
For test data: We could add a script in `examples.py` that creates artificial 2D/4D data, or as a pytest fixture.

**Comment by adehecq** [comment]
> For documenting: I would add a "command line user guide" like in Rasterio (but in the Documentation section, not as its own header): https://rasterio.readthedocs.io/en/stable/cli.html 

So you mean, I should have a subsection in the "Reference" section, e.g. after the API?
Can the documentation of the function be parsed automatically like for the API? If I copy paste the api.rst file and update to list geoviewer.py, would that work?

> For test data: We could add a script in `examples.py` that creates artificial 2D/4D data, or as a pytest fixture.

Right, yes we could have this in the pytest fixture, as it does not make much sense to store the artificial data.

**Comment by rhugonnet** [comment]
Yes, exactly! A `Command-line interface` section in `Reference`. 

To automatically document the function, looks like we can add the `sphinx-argparse` module: https://sphinx-argparse.readthedocs.io/en/latest/usage.html#basic-usage. The basic usage looks fairly easy. And if we move to `click` at some point instead of `argparse`, we can do the same with `sphinx-click`: https://sphinx-click.readthedocs.io/en/latest/! :slightly_smiling_face:

**Comment by adehecq** [comment]
I've managed to build the CLI documentation locally, but it fails on readthedocs (will be visible at [this link](https://geoutils-adehecq.readthedocs.io/en/fix_geoviewer/cli.html)).

One issue is that the usage window is too large for the width so we need to scroll to see the whole usage. I tried the solutions given at this [link](https://sphinx-argparse.readthedocs.io/en/latest/misc.html) without success. I tried :
- adding the `docutils.conf` file as suggested (2nd option)
- adding the suggested lines in the `doc/source/_static/css/custom.css` file (1st option) 

Maybe I missed something...

**Comment by rhugonnet** [comment]
For your bug: looks like a `distutils` error due to Python 3.12 like Andrew's issue: #428.
Adding `python >=3.9, <3.12` to the `dev-environment` and normal `environment` could solve the issue, and you could close that issue as well!

**Comment by rhugonnet** [comment]
Maybe your PR is also the occasion to fix #430, should be just a couple lines :wink:. The current default interpolation degrades the output of the image viewer quite a bit by propagating NaNs.

**Comment by rhugonnet** [comment]
For #430, you probably just need to add this at the start of `Raster.show()`:

```python
 if "interpolation" not in kwargs.keys():
    kwargs.update({"interpolation": "None"})
```
(the "" around None being very important!! you probably know this annoying matplotlib behaviour... :sweat_smile: )

File: dev-environment.yml [hunk 1]
```diff
@@ -35,6 +35,7 @@ dependencies:
   - sphinx-design
   - sphinx-autodoc-typehints
   - sphinxcontrib-programoutput
+  - sphinx-argparse
```
[the remark this answers is not in the document]
Comment (adehecq) [reply dev-environment.yml:38]: Sorry, I missed that info in the `dev-environment.yml` file !

**Comment by adehecq** [comment]
Finally managed to get the documentation to build properly! The result can bee seen here for the moment: https://geoutils-adehecq.readthedocs.io/en/fix_geoviewer/cli.html

The issue was regarding the path to the script `geoviewer.py`. If using a relative path, it depends where the sphinx command is ran from. For readthedocs, it seems like "../../bin/geoviewer.py" works, but it fails in `test_docs.py` because the command is run in the root folder.
Therefore, using simply "geoviewer.py" works better as the script is added to the path by `pip install`, but is a bit ambiguous. In the future, it would be simpler to use the "module" argument of sphinx-argparse, but require the tool to be located in a module, or use sphinx-click.

**Comment by rhugonnet** [comment]
Perfect! So it works locally + CI with just `geoviewer`? I think that is perfect, this is why we go through the trouble of declaring the module properly through `setuptools` :slightly_smiling_face: 

> That's a good idea and a quick fix. However, on the longer term, I was wondering if we should not have default changes to the rcParams like what is done in Seaborn?

Not sure I see exactly, but if you think it is useful it could be an issue for future improvement!

**Comment by adehecq** [comment]
> Not sure I see exactly, but if you think it is useful it could be an issue for future improvement!

Added issue #438.

**Pull request merged**